common/settings/settings.ts

```diff
@@ -30,6 +30,96 @@ export interface MiscSettings {
+they don't ever collect the lemma).
+
+Lemma In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |         MATCH        |
+| Inflection      |          NO          |          NO          |
+-----------------------------------------------------------------
+
+Inflection In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |          NO          |
+| Same Inflection |          NO          |         MATCH        |
+| Diff Inflection |          NO          |          NO          |
+-----------------------------------------------------------------
+*/
+export enum TokenMatchStrategy {
+    ANY_FORM_COLLECTED = 'ANY_FORM_COLLECTED', // All scenarios above result in MATCH
+    LEMMA_OR_EXACT_FORM_COLLECTED = 'LEMMA_OR_EXACT_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED and EXACT_FORM_COLLECTED columns above
+    LEMMA_FORM_COLLECTED = 'LEMMA_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED column above
+    EXACT_FORM_COLLECTED = 'EXACT_FORM_COLLECTED', // See EXACT_FORM_COLLECTED column above
+}
+
+export enum TokenMatchStrategyPriority {
+    EXACT = 'EXACT',
+    LEMMA = 'LEMMA',
+    BEST_KNOWN = 'BEST_KNOWN',
+    LEAST_KNOWN = 'LEAST_KNOWN',
+}
+
+export enum DictionaryAnkiTreatSuspended {
+    NORMAL = 'normal',
+    MATURE = 'mature',
+    YOUNG = 'young',
+    UNKNOWN = 'unknown',
+}
+
+export enum TokenStyling {
+    TEXT = 'text',
+    UNDERLINE = 'underline',
+    OVERLINE = 'overline',
+}
+
+export enum TokenStatus {
+    UNCOLLECTED = 0,
+    UNKNOWN = 1,
+    YOUNG = 2,
+    MATURE = 3, // If ever adding more statues, they should go after MATURE and getFullyKnownTokenStatus should be updated
+}
+
+export function getFullyKnownTokenStatus(): TokenStatus {
+    return TokenStatus.MATURE; // If future statuses are optional, this logic may need to change
+}
+
+export interface DictionarySubtitleAppearance {
+    readonly tokenStyling: TokenStyling;
+    readonly tokenStylingThickness: number;
+    readonly colorizeFullyKnownTokens: boolean;
+    readonly tokenStatusColors: string[]; // Indexed by TokenStatus
+}
+
+export interface DictionaryTrack {
+    readonly dictionaryColorizeOnVideo: boolean;
+    readonly dictionaryColorizeOnApp: boolean;
+    readonly dictionaryTokenMatchStrategy: TokenMatchStrategy;
+    readonly dictionaryTokenMatchStrategyPriority: TokenMatchStrategyPriority;
+    readonly dictionaryYomitanUrl: string;
+    readonly dictionaryYomitanScanLength: number;
+    readonly dictionaryAnkiEnabled: boolean;
+    readonly dictionaryAnkiConnectUrl: string;
```

Comment:
I think it would be very rare for a user to want to use a different AnkiConnect URL then the one that's already configured. I know that the defaults will be the same and most users won't change this, but I don't think we need to introduce a new field either.